test/TestApps/PushNotificationsWin32App/main.cpp

```diff
@@ -1,12 +1,70 @@
 ﻿#include "pch.h"
 #include <iostream>
+#include <sstream>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.Security.Cryptography.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <winrt/Windows.Web.Http.Headers.h>
+
 using namespace winrt;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
 using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::ApplicationModel::Activation;
 using namespace winrt::Microsoft::ProjectReunion;
+using namespace Windows::Web::Http;
+
+std::wstring RequestChannel()
+{
+    return // Insert ChannelUri here.;
+}
+
+std::wstring BuildNotificationPayload()
+{
+    std::wstring channel = RequestChannel();
+    std::wstring channelUri = L"\"ChannelUri\":\"" + channel + L"\"";
+    std::wstring x_wns_type = L"\"X_WNS_Type\": \"wns/raw\"";
+    std::wstring contentType = L"\"Content_Type\": \"application/octet-stream\"";
+    std::wstring payload = L"\"Payload\": \"<toast></toast>\"";
+    std::wstring delay = L"\"Delay\": \"false\"";
+    return { L"{" + channelUri + L"," + x_wns_type + L"," + contentType + L"," + payload + L"," + delay + L"}" };
+}
+
+void SendRequestToServer()
+{
+    HttpResponseMessage httpResponseMessage;
+    std::wstring httpResponseBody;
+    try {
+        // Construct the HttpClient and Uri. This endpoint is for test purposes only.
+        HttpClient httpClient;
+        Uri requestUri{ L"https://notificationtestappserver.azurewebsites.net/api/PostPushNotification?" };
+
+        // Construct the JSON to post.
+        HttpStringContent jsonContent(
+            BuildNotificationPayload(),
+            UnicodeEncoding::Utf8,
+            L"application/json");
+
+        // Post the JSON, and wait for a response.
+        httpResponseMessage = httpClient.PostAsync(
+            requestUri,
+            jsonContent).get();
+
+        // Make sure the post succeeded, and write out the response.
+        httpResponseMessage.EnsureSuccessStatusCode();
+        httpResponseBody = httpResponseMessage.Content().ReadAsStringAsync().get();
+        std::wcout << httpResponseBody.c_str() << std::endl;
+    }
+    catch (winrt::hresult_error const& ex)
+    {
+        std::wcout << ex.message().c_str();
+    }
+}
 
 int main()
 {
-    // Add your code here.
-    std::cout << "Hello World!\n";
+    bool succeeded = true;
+    SendRequestToServer();
+    std::getchar();
     return 0;
```

Comment:
IMHO, main should return a failure if the Send operation fails. Otherwise it should return success.